pkg/cloud/cloud_test.go

```diff
@@ -338,6 +338,11 @@ func TestRenderedResources(t *testing.T) {
 				checkLeaderElection(t, podSpec)
 				checkCloudControllerManagerFlags(t, podSpec)
 				checkTrustedCAMounted(t, podSpec)
+				if platform.platformStatus.Type != configv1.VSpherePlatformType {
+					// we don't use service account credentials on vSphere as its CCM behavior
+					// is different from the in-tree cloud provider.
```

Comment:
Can you explain in the PR description why this is the case, what does it do differently, would be good to know for the future why we have this exception

Reply:
Done